src/middlewares/isBought.js

```diff
@@ -0,0 +1,43 @@
+import JwtUtil from "../utils/generateToken";
+import { productOrder, User } from "../database/models/index";
+
+const isBought = async (req, res, next) => {
+  try {
+    const { id } = req.params;
```

Comment:
> You don't need these codes here. We already have the data of the user who has logged in. We have a specific middleware for that.

this middleware is to check if the product is already bought or not and also check if the buyer who is trying to give a feedback is the one who actually bought the product.  so i don't understand why i get rid of it while i still need to check if the product is bought because in the pivotaltracker the task is to allow buyers to give feedback on only product they have bought

Reply:
Why would you need to decode a token again when you have checked the logged in user and the user role?

Reply:
yeah i get it let me work on it and modify it